app/assets/javascripts/services/urlService.js

```diff
@@ -0,0 +1,22 @@
+const getHash = key => {
+  switch (key) {
+    case 'story':
+      const match = window.location.hash.match(/#story-(\d+).*/);
+
+      if(match) {
+        return parseInt(match[1]) || null;
+      }
+
+      return null
+    default:
+      const [, rest] = window.location.hash.split(`${key}=`)
+
+      if (rest) {
+        return rest.split('&')[0]
+      }
+
+      return null
+  }
+}
+
```

Comment:
What you think about we handling this with another approach so we have a more readable code?? Also, code climate is failing in this file, because of it complexity.

Reply:
Yeah, I percepted this too, I will refactor this file.